pkg/asset/installconfig/azure/validation.go

```diff
@@ -995,3 +1000,65 @@ func checkBootDiagnosticsURI(client API, diag *aztypes.BootDiagnostics, region s
 	}
 	return nil
 }
+
+func validateSubnetNatGateway(client API, fieldPath *field.Path, subnet *aznetwork.Subnet, outboundType aztypes.OutboundType, role capz.SubnetRole, resourceGroup, virtualNetwork string) field.ErrorList {
+	var allErrs field.ErrorList
+	if outboundType != aztypes.NATGatewayMultiZoneOutboundType && outboundType != aztypes.NATGatewaySingleZoneOutboundType {
+		return allErrs
```

Comment:
return append(allErrs, field.Invalid(fieldPath, outboundType, "invalid outbound type specified"))
allErrs doesn't have any errors in it at this point.

Reply:
This validation only runs for those 2 conditions. It should skip this function if like the input is `LoadBalancer`. Felt like the calling function had a lot of these check so moved it here.